sips/sip-028/sip-028-sbtc_peg.md

```diff
@@ -0,0 +1,217 @@
+
+#### Deposits
+
+The main steps of the sBTC Deposit flow will be as follows:
+
+1. **Deposit request:** A bitcoin holder creates a transaction on Bitcoin.
+2. The deposit transaction contains a UTXO (deposit UTXO) spendable by sBTC Signers, with an OP_DROP payload.
+3. The payload contains the recipient address of the sBTC among other relevant info for the deposit.
+   - The relevant info could contain a fee suggestion or max_fee.
+4. **Proof of deposit:** The bitcoin holder submits a proof of deposit on Stacks by invoking the Signer binary API.
+5. **Deposit accept:** 
+6. **Deposit redeem:** The sBTC Signers redeem the deposit by consuming the deposit UTXO, consolidating it into the sBTC UTXO.
+7. **Mint:** The sBTC Signers finalize the deposit acceptance making a Clarity contract call that mints the sBTC on the Stacks Layer.
+
+#### Withdrawals (Redeeming sBTC)
+
+The main steps of the sBTC withdrawal flow are as follows:
+
+1. **Withdrawal request:** An sBTC holder calls the `withdraw-request` function in the `.sbtc` contract.
+   - This transfers the requested amount of sBTC to the `.sbtc` contract & mints the user a non-transferable locked-sBTC as a placeholder.
+2. **Withdrawal accept:** If accepted, the following happens:
+   - The signers create a transaction on Bitcoin which returns the requested amount to the designated address.
+   - Once the Bitcoin transaction is confirmed, the signers make a smart contract call to one of the `.sbtc` contracts to mark the transaction as fulfilled.
+   - If successful, the resulting Stacks transaction will record the withdrawal request as complete & will accordingly burn the user’s locked-sBTC.
+3. **Withdrawal reject:** If instead the request is rejected, the sBTC signers will call the `withdraw-reject` function in the `.sbtc` smart contract. This function does the following:
+   - Returns the sBTC to the holder.
+   - Records the signer votes.
+
+### Auxiliary Features
+
+Auxiliary features of the sBTC protocol are described below.
+
+#### Stacks Transaction Fee Sponsorship
+
+sBTC will include the option to have sBTC transactions on Stacks be sponsored in return for some sBTC. Using the approach suggested in this [issue](https://github.com/stacks-network/stacks-core/issues/4235), sBTC users will be able to pay for their transaction fees in sBTC with support from an existing STX holder, provided the wallet supports it. The proposed solution introduces atomic transaction bundles on Stacks, which enable sBTC payments to sponsors for covering STX transaction fees. STX is maintained as the sole gas token, but the user only has to interact with sBTC.
+
+#### Signer Key Rotation
+
+Mechanisms are provided for the scenario where a signer wants to rotate their key. For this to happen, signers must coordinate offline and vote on-chain on the new signer set (aka set of keys). Once the new signer set is determined, the signers conduct a wallet handoff and re-execute the key generation event.
```

Comment:
What mechanisms, specifically?

Reply:
removed in 321d118f5dd48f1f526bd6d9cfd89497878ac98b